### PR TITLE
Add linearHBmodel to implement 1D IMEX for vertical viscosity.

### PR DESCRIPTION
**Pull request by blallen**
This PR adds a LinearHBModel to implement the 1D IMEX for vertical viscosity and diffusion. 

The configuration is only tested for viscosity as this is constant in space and time. We need to modify the linear solvers to update the matrices with time to account for the diffusion changes required for the convective adjustment.

### CODE REVIEW

**Comment by lcw** [comment]
Not sure what you want me to review.  I took a quick look and it seems reasonable but I don't know exactly what you are trying to do.

Note that rebuilding and factoring the column matrices is expensive.  There is a branch https://github.com/climate-machine/CLIMA/tree/rm/columnwise-gmres
being worked on by @rohanmclure that should provide a matrix free approach.

**Comment by blallen** [comment]
> Not sure what you want me to review. I took a quick look and it seems reasonable but I don't know exactly what you are trying to do.

Mostly check that things make sense. The main goal of this PR is to compute the terms for vertical viscosity and diffusion implicitly. The ARK/SingleColumnLU method works for viscosity and that is what has been tested so far. 

> Note that rebuilding and factoring the column matrices is expensive. There is a branch https://github.com/climate-machine/CLIMA/tree/rm/columnwise-gmres
> being worked on by @rohanmclure that should provide a matrix free approach.

This is the next step; implement something that works for a time-dependent diffusivity. The GMRES branch looks useful. I'm currently looking through the LinearSolver and ODESolver portions of the code to understand how they interact. For instance, can the ARK solvers use both the iterative and LU solvers for the linear solve?

I'm not quite sure if I want to merge this PR before we get things working for the diffusivity.

File: test/Ocean/Hydrostatic_Boussinesq/simple_box_IMEX.jl [hunk 1]
```diff
@@ -87,6 +87,57 @@ function ocean_init_state!(P::SimpleBox, Q, A, coords, t)
   Q.θ = 20 # 9 + 8z/H
 end
 
+function make_callbacks(vtkpath, step, nout, mpicomm, odesolver, dg, model, Q)
+  if isdir(vtkpath)
+    rm(vtkpath, recursive=true)
+  end
+  mkpath(vtkpath)
+  mkpath(vtkpath*"/weekly")
+  mkpath(vtkpath*"/monthly")
```
Comment (charleskawczynski) [inline test/Ocean/Hydrostatic_Boussinesq/simple_box_IMEX.jl]:
```
  mkpath(joinpath(vtkpath,"weekly"))
  mkpath(joinpath(vtkpath,"monthly"))
```

**Comment by blallen** [comment]
bors try

**Comment by blallen** [comment]
bors try

**Comment by kpamnany** [comment]
@blallen: I suggest we start squashing commits before merging future PRs. This allows master to have a cleaner commit history and also, each commit on master will work. [Here](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes) is a nice description of how it's done.

**Comment by blallen** [comment]
@kpamnany you're in luck. that build failed and I had the chance to squash everything to a single commit

**Comment by blallen** [comment]
also @kpamnany, after this PR is merged, I will get up a PR with the ocean model using the config+driver setup. starting to clean up these scripts and realized it would be easier to just use your interface at this point

**Pull request merged**